src/encoded/audit/experiment.py

```diff
@@ -57,162 +57,177 @@ def audit_experiment_chipseq_control_read_depth(value, system, files_structure):
+                            and
+                            derived_from.get('dataset') in controls_files_structures
+                            and
+                            check_pipeline('ChIP-seq read mapping',
+                                           derived_from.get('@id'),
+                                           controls_files_structures[derived_from.get('dataset')]))]
+                    control_bam_details = []
+                    cumulative_read_depth = 0
+                    for bam_file in derived_from_external_bams:
+                        control_depth = get_chip_seq_bam_read_depth(bam_file)
+                        control_target = get_control_target_name(
+                            bam_file.get('dataset'),
+                            control_objects)
+                            
+                        if control_target not in ['Control-human', 'Control-mouse']:
+                            detail = (
+                                'Control {} file {} ' +
+                                'has a target {} that is neither ' +
+                                'Control-human nor Control-mouse.').format(
+                                    bam_file['output_type'],
+                                    bam_file['@id'],
+                                    control_target)
+                            yield AuditFailure('inconsistent target of control experiment', detail, level='WARNING')
+                            return
+                        if control_depth and control_target:
+                            cumulative_read_depth += control_depth
+                            triple = (
+                                bam_file.get('@id'),
+                                control_depth,
+                                bam_file.get('dataset'))
+                            control_bam_details.append(triple)
+                    yield from check_control_read_depth_standards(
+                        peaks_file.get('@id'),
+                        peaks_file.get('assembly'),
+                        cumulative_read_depth,
+                        control_bam_details,
+                        target_name,
+                        target_investigated_as)
+
+
+def get_control_target_name(control_id, control_objects):
+    control = control_objects.get(control_id)
+    if control and 'target' in control and \
+       'name' in control['target']:
+        return control['target']['name']
+    return False
+
+
+def check_pipeline(pipeline_title, control_file_id, file_structure):
+    control_file = file_structure.get('alignments')[control_file_id]
+    if ('analysis_step_version' in control_file and
+            'analysis_step' in control_file.get('analysis_step_version')):
+        pipelines = control_file.get('analysis_step_version').get('analysis_step').get('pipelines')
+        return pipeline_title in get_pipeline_titles(pipelines)
+    return False
```

Comment:
Again we don't need a return false because the default is return None, which works in our case.  In general, functions should be boolean, where they return True or False, or they should return something or nothing.

Reply:
require re-demo